Bottom: 208f7ed73cbd38b0a5da0a22cf473eddfbd04558
Top:    3b95bd024d25f22c5b1594e23e80e115e576dc05
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-08-14 15:30:30 -0500

Configure freenode as IRC reporter


---
diff --git a/ansible-playbook/group_vars/all/credentials.yml b/ansible-playbook/group_vars/all/credentials.yml
index 78b584d2..3def5922 100644
--- a/ansible-playbook/group_vars/all/credentials.yml
+++ b/ansible-playbook/group_vars/all/credentials.yml
@@ -2,6 +2,7 @@
 # passwords
 buildbot_admin_password:             '{{ lookup("password", playbook_dir + "/credentials/buildbot/admin"                      , length=20) }}'
 buildbot_email_password:             '{{ lookup("password", playbook_dir + "/credentials/buildbot/buildbot@novatech-llc.com"  , length=20) }}'
+buildbot_irc_password:               '{{ lookup("password", playbook_dir + "/credentials/buildbot/irc-NovaTechCI"             , length=20) }}'
 buildbot_worker_ntel_password:       '{{ lookup("password", playbook_dir + "/credentials/buildbot/worker_ntel"                , length=20) }}'
 buildbot_worker_ptxdist_password:    '{{ lookup("password", playbook_dir + "/credentials/buildbot/worker_ptxdist"             , length=20) }}'
 buildbot_webhook_secret:             '{{ lookup("password", playbook_dir + "/credentials/buildbot/webhook"                    , length=20) }}'
diff --git a/docker/buildbot-master/Dockerfile b/docker/buildbot-master/Dockerfile
index a79778ef..d9b8f54b 100644
--- a/docker/buildbot-master/Dockerfile
+++ b/docker/buildbot-master/Dockerfile
@@ -19,7 +19,8 @@ RUN pip --no-cache-dir install \
     'buildbot-grid-view' \
     'buildbot-waterfall-view' \
     'buildbot-www' \
-    'ldap3'
+    'ldap3' \
+    'pyOpenSSL'
 
 # Create buildbot user
 ARG BUILDBOT_UID=1000
diff --git a/docker/buildbot-master/buildbot/master.cfg b/docker/buildbot-master/buildbot/master.cfg
index 5f1c821e..1ac46a1a 100644
--- a/docker/buildbot-master/buildbot/master.cfg
+++ b/docker/buildbot-master/buildbot/master.cfg
@@ -5,7 +5,8 @@
 # 'master.cfg' in your buildmaster's base directory.
 
 import os
-from buildbot.plugins import secrets,steps,util
+import json
+from buildbot.plugins import reporters, secrets, steps, util
 from urlparse import urlparse
 
 os.environ['LONG_RUN_TIMEOUT']='14400'
@@ -34,10 +35,17 @@ c['secretsProviders'] = [secrets.SecretInAFile(dirname="/run/secrets")]
 
 c['services'] = []
 
-from buildbot.plugins import reporters
-irc = reporters.IRC(host="172.16.64.3",
-                    nick="orionbb",
-                    channels=["#orion"])
+irc = reporters.IRC(
+    allowForce=False,
+    allowShutdown=False,
+    channels=json.loads(os.getenv("BUILDBOT_IRC_CHANNELS")),
+    host=os.getenv("BUILDBOT_IRC_HOST"),
+    nick=os.getenv("BUILDBOT_IRC_NICK"),
+    notify_events=[],
+    password=os.getenv("BUILDBOT_IRC_PASS"),
+    useColors=True,
+    useSSL=True
+)
 c['services'].append(irc)
 
 mn = reporters.MailNotifier(
